File: sdks/java/src/main/java/com/boltprediction/sdk/model/WorkingConditionSchema.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 工况信息 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class WorkingConditionSchema {

    @JsonProperty("temperature")
    private Object temperature;

    @JsonProperty("pressure")
    private Object pressure;

    @JsonProperty("humidity")
    private Object humidity;

    @JsonProperty("vibration")
    private Object vibration;

    @JsonProperty("load_condition")
    private Object loadCondition;

    @JsonProperty("operating_hours")
    private Object operatingHours;

    @JsonProperty("maintenance_cycle")
    private Object maintenanceCycle;

    @JsonProperty("last_maintenance_date")
    private Object lastMaintenanceDate;

    @JsonProperty("equipment_age")
    private Object equipmentAge;

    @JsonProperty("extra")
    private Object extra;

    public Object getTemperature() {
        return temperature;
    }

    public void setTemperature(Object temperature) {
        this.temperature = temperature;
    }

    public Object getPressure() {
        return pressure;
    }

    public void setPressure(Object pressure) {
        this.pressure = pressure;
    }

    public Object getHumidity() {
        return humidity;
    }

    public void setHumidity(Object humidity) {
        this.humidity = humidity;
    }

    public Object getVibration() {
        return vibration;
    }

    public void setVibration(Object vibration) {
        this.vibration = vibration;
    }

    public Object getLoadCondition() {
        return loadCondition;
    }

    public void setLoadCondition(Object loadCondition) {
        this.loadCondition = loadCondition;
    }

    public Object getOperatingHours() {
        return operatingHours;
    }

    public void setOperatingHours(Object operatingHours) {
        this.operatingHours = operatingHours;
    }

    public Object getMaintenanceCycle() {
        return maintenanceCycle;
    }

    public void setMaintenanceCycle(Object maintenanceCycle) {
        this.maintenanceCycle = maintenanceCycle;
    }

    public Object getLastMaintenanceDate() {
        return lastMaintenanceDate;
    }

    public void setLastMaintenanceDate(Object lastMaintenanceDate) {
        this.lastMaintenanceDate = lastMaintenanceDate;
    }

    public Object getEquipmentAge() {
        return equipmentAge;
    }

    public void setEquipmentAge(Object equipmentAge) {
        this.equipmentAge = equipmentAge;
    }

    public Object getExtra() {
        return extra;
    }

    public void setExtra(Object extra) {
        this.extra = extra;
    }

}

File: sdks/python/bolt_prediction_sdk/models/working_condition_schema.py
"""
WorkingConditionSchema 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class WorkingConditionSchema(SDKBaseModel):
    """工况信息"""

    temperature: Optional[Any] = Field(description="环境温度", default=None)
    pressure: Optional[Any] = Field(description="系统压力", default=None)
    humidity: Optional[Any] = Field(description="环境湿度", default=None)
    vibration: Optional[Any] = Field(description="振动水平", default=None)
    load_condition: Optional[Any] = Field(description="负载状况 light/medium/heavy/overload", default=None)
    operating_hours: Optional[Any] = Field(description="运行时长（小时）", default=None)
    maintenance_cycle: Optional[Any] = Field(description="维护周期", default=None)
    last_maintenance_date: Optional[Any] = Field(description="上次维护日期", default=None)
    equipment_age: Optional[Any] = Field(description="设备使用年限", default=None)
    extra: Optional[Any] = Field(description="其他工况参数", default=None)
